packages/core/certora/optimistic-asserter/specs/Asserter_Auxiliary.spec

```diff
@@ -0,0 +1,186 @@
+// For every new spec you create, import this basic spec file:
+import "./Asserter_Base.spec"
+import "./dispatchedMethods.spec"
+
+using TestnetERC20 as testERC20
+/**************************************************
+ *      Top Level Properties / Rule Ideas         *
+ **************************************************/
+// Here you can write your ideas for rules and implement them 
+// in this spec.
+
+/**************************************************
+ *                  Misc. rules                   *
+ **************************************************/
+
+// A simple rule that checks which of the main contract methods
+// are reachable (reach the assert false statement after function call).
+ rule sanity(method f) {
+    env e;  // Environment variable - includes all transaction and block information.
+    calldataarg args; // arbitrary calldata - adapted for every method signature implicitly
+    f(e, args);
+    assert false;
+}
+
+// Checks that view functions in the contract never revert. 
+rule viewFuncsDontRevert(method f) 
+filtered{f -> f.isView} { // filters the methods to only view functions
+    env e;
+    require e.msg.value == 0;
+    calldataarg args;
+    f@withrevert(e, args);
+
+    assert !lastReverted;
+}
+
+/**************************************************
+ *                 "Who changed" Rules            *
```

Comment:
I think these are also expected to fail. How should these be used? Do we pre-filter methods that we _expect_ to change that variable? Or are these just expected to fail, so we shouldn't run in ci?

Reply:
Yes, the purpose of these rules is done by failure detection: the functions that change the observed variable will cause these rules to fail and then we can monitor them. It's mainly useful for someone who doesn't understand the functionality of the system very well, so these rules help him detect when do the state variables change.
It shouldn't be a part of CI, it's merely an auxiliary spec.

Reply:
Understood.